x-pack/plugins/security_solution/public/entity_analytics/pages/entity_analytics_management_page.tsx

```diff
@@ -5,38 +5,174 @@
+    localStorage.setItem('dateEnd', end);
+  };
+
+  const calculateNextRunTime = useCallback((): string => {
+    if (runEngineEnabled) {
+      const currentTime = new Date();
+      const runAtTime = riskEngineStatus?.risk_engine_task_status?.runAt
+        ? new Date(riskEngineStatus.risk_engine_task_status.runAt)
+        : new Date();
+      const minutesUntilNextRun = Math.round((runAtTime.getTime() - currentTime.getTime()) / 60000);
+      return `Next engine run in ${minutesUntilNextRun} minutes`;
+    }
+    return '';
+  }, [runEngineEnabled, riskEngineStatus]);
+
+  useEffect(() => {
+    let intervalId: string | number | NodeJS.Timeout;
+
+    // Poll only if the engine is in the 'running' state
+    if (riskEngineStatus?.risk_engine_task_status?.status === 'idle') {
+      setNextRunTime(calculateNextRunTime()); // Initial calculation
+
+      intervalId = setInterval(() => {
+        setNextRunTime(calculateNextRunTime());
+      }, 60000); // Poll every minute
+    }
+
+    return () => {
+      if (intervalId) {
+        clearInterval(intervalId);
+      }
+    };
+  }, [calculateNextRunTime, riskEngineStatus?.risk_engine_task_status?.status]);
+
   return (
     <>
       <RiskEnginePrivilegesCallOut privileges={privileges} />
       <EuiPageHeader
         pageTitle={
-          <EuiFlexGroup>
+          <EuiFlexGroup alignItems="center" justifyContent="spaceBetween">
             <EuiFlexItem data-test-subj="entityAnalyticsManagementPageTitle" grow={false}>
               {ENTITY_ANALYTICS_RISK_SCORE}
             </EuiFlexItem>
-            <EuiBetaBadge label={BETA} size="s" />
+
+            <EuiFlexItem grow={false}>
+              <EuiFlexGroup alignItems="center" gutterSize="s">
+                {/* Run Engine Button */}
+                {runEngineEnabled && (
+                  <EuiFlexItem grow={false}>
+                    <EuiButton
+                      size="s"
+                      iconType="play"
+                      isLoading={isLoading}
+                      onClick={async () => {
+                        await handleRunEngineClick();
+                        if (!isLoading) {
+                          addSuccess(i18n.RISK_SCORE_ENGINE_RUN_SUCCESS, { toastLifeTimeMs: 5000 });
+                        }
+                      }}
+                    >
+                      {'Run Engine'}
+                    </EuiButton>
+                  </EuiFlexItem>
+                )}
+
+                {/* Vertical Line */}
+                {runEngineEnabled && (
+                  <div
+                    className="vertical-line"
+                    style={{ height: '24px', borderLeft: '1px solid #ccc', margin: '0 8px' }}
+                  />
+                )}
+
+                {/* Text: "Next engine run in {} minutes" */}
+                {runEngineEnabled && (
+                  <EuiFlexItem grow={false}>
+                    <span style={{ fontSize: '14px', color: '#888888', fontWeight: 'normal' }}>
```

Comment:
We must avoid using colours and font sizes like here and also using style. 
We should also avoid injecting CSS via `style`.

The `Eui` library has components and variables to help us. If that isn't enough, we can use CSS with `emotion` but always try to use variables from the `theme`.

For this case you probably need to use `EuiText` like `<EuiText size="s" color="danger">`

This also applies for line 133

EuiText: https://eui.elastic.co/#/display/text

Reply:
Of course! Great point, @machadoum ,thank you! I’ll make the change, including updating the small vertical line. I hadn’t found an EUI component or `EuiIcon` that serves as a vertical line, but I agree using `emotion` is a good solution. I’ll implement this change here and elsewhere as needed. Thanks again!